azurelinuxagent/common/protocol/util.py

```diff
@@ -319,8 +309,7 @@ def clear_protocol(self):
 
     def get_protocol(self, by_file=False):
         """
-        Detect protocol by endpoints, if by_file is True,
```

Comment:
we need to remove the unit tests as well (test_metadata.py)

Reply:
https://travis-ci.org/Azure/WALinuxAgent/jobs/635512763 would give you the places metadata is used. Would help in removing the metadata protocol based unittests.

Reply:
Done, as well as mockmetadata.py

Reply:
Searching for azurelinuxagent.common.protocol.metadata would point you to other places where metadata is used.